packages/scripts.nix

```
@@ -531,4 +535,32 @@
     ];
     text = builtins.readFile ./test-peerpods.sh;
   };
+
+  # Sync the go directive between go.mod/go.work files (that is, the 'go' statement of these files).
+  # We take take the latest version we find and use that everywhere.
+  go-directive-sync = writeShellApplication {
+    name = "go-directive-sync";
+    runtimeInputs = with pkgs; [
+      go
+      findutils
+      coreutils
+    ];
+    text = ''
+      set -euo pipefail
+
+      modFiles=$(find . -regex '.*\(go.mod\|go.work\)$')
+
+      goVers=()
+      while IFS= read -r f; do
+        ver=$(grep '^go [0-9]\+\.[0-9]\+\.[0-9]\+$' "$f")
```

```suggestion
        ver=$(grep -E '^go [0-9]+[.][0-9]+[.][0-9]+$' "$f")
```

* Use `-E` for unescaped `+` support.
* Nit - I find it harder to reason about backslash-escaped dots than `[.]` because I'm always worried about the shell hijacking the escape.